pkgs/development/tools/sumneko-lua-language-server/default.nix

```diff
@@ -65,6 +65,7 @@ stdenv.mkDerivation rec {
 
     makeWrapper "$out"/share/lua-language-server/bin/lua-language-server \
       $out/bin/lua-language-server \
+      --run 'export XDG_CACHE_HOME="''${XDG_CACHE_HOME:-$HOME/.cache}"' \
       --add-flags "-E $out/share/lua-language-server/main.lua \
       --logpath='\$XDG_CACHE_HOME/sumneko_lua/log' \
```

Comment:
I know it is not a part of this PR, but I am not a fan of using single quotes here. Variables quoted this way are not expanded by the shell. I would replace single quotes `'` with escaped double qotes `\"`

Reply:
This actually ended up being required in the current version, since sumneko-lua will only expand the path once (relevant code: <https://github.com/sumneko/lua-language-server/blob/master/script/utility.lua#L665>). Therefore, it would just create a `\~` directory.